pandas/core/series.py

```diff
@@ -495,7 +495,9 @@ def _init_dict(
         elif index is not None:
             # fastpath for Series(data=None). Just use broadcasting a scalar
             # instead of reindexing.
-            values = na_value_for_dtype(pandas_dtype(dtype), compat=False)
```

Comment:
why did you get rid of the `pandas_dtype`?

Reply:
that was following on from https://github.com/pandas-dev/pandas/pull/44615#r790870110

Reply:
that link isn't working for me.  can you describe the comment?

Reply:
no worries, it's a review above on the 25th January

> this should be unnecessary. the annotation says we have DtypeObj | None. If the annotation is wrong, we need to fix that before getting here